### ISSUE TITLE
[XFCE] desktop settings stuck with french locales on wallpaper loading

### ISSUE THREAD

**Issue by philmmanjaro** [manjaro/release-plan#183]
<img src="https://forum.manjaro.org/uploads/default/optimized/3X/f/d/fd67733260754813df971d6d13fe06bf8d9dcfd7_1_678x500.png">

**ISSUE**

Desktop settings wallpaper change is somehow broken when having **breath-wallpaper** installed.

**FINDING**

I tested this now further. It doesn’t matter which format the background has. I added now the same breath.png saved as .jpg and got a blocked settings application. Then I renamed it from a.jpg till m.jpg. It all blocked it. Starting with n.jpg till z.jpg it worked by German alphabetical order. Also pictures with 0.jpg - 9.jpg don’t work. Also starting with special characters doesn’t work.

If you however start with a letter beginning with n and later it works. So loading a picture called na.jpg works. However a picture called an.jpg not.

So please [open a ticket upstream](https://bugzilla.xfce.org/) and include my findings. This is only related to French.

**WORKAROUND**

* Don't install Wallpapers having only letters with a-m (German Alphabet) in them, when not having at least 30% letters from n-z in them. Seems only affecting French.
* Use `/usr/share/backgrounds/xfce` as default folder.

**Comment by FredBezies** [comment]
I don't want to start world war III, but I think French is the fifth spoken language in the world after chinese, hindi, english and spanish.

It is really bad to see you don't seem to care about what could be around 15 or 20% of your users. That's annoying to stay polite.

You're providing a workaround. Good. But is it not a bug in your theme which activates one in Xfce?

I can fix your bug by not using your provided theme. So, for me it is a bug in your default theme. Maybe 18.0 official theme won't be plagued by this bug.

Have a good day, even if you don't like french. I hate german because of a teacher who burped while speaking it to us.

**Comment by philmmanjaro** [comment]
@FredBezies: I don't see an issue with the Theme `Vertex-Maia` here. Even if I switch it to `Vertex-Maia-Dark` and then try change the backgrounds it stuck. You can make it even stuck with the combination `Adwaita` and `breath.png`.

What also works is only having **breath.png** installed in `/usr/share/backgrounds/` or all pointing to `/usr/share/backgrounds/xfce/breath.png`

Also having pictures called `breath.png` and `breath[1-6].png` works also in folder `/usr/share/backgrounds/`.
Maybe another language might have the same issue, but only French is currently known to us.

So in other words: There are situation of combinations which simply don't work with French:

* several themes, not fixed to `Vertex-Maia`, don't work with pictures in `/usr/share/backgrounds/` starting with `a-m` by German alphabetical order not having at least 30% of the letters `n-z` in them.
* kinda works with `/usr/share/backgrounds/xfce` as default folder

So I don't know why I should make a fuss about French and prioritize it, when other languages just work fine. This is a general issue with the language French and given combinations. So compile the xfce-settings package with debug headers and start a git-bisect to find the commit which might created a regression for given language by analysing the issues the debugger might present you. As of now I've no time to invest in this matter from my end.

**Comment by FredBezies** [comment]
Fuss? Only **15 to 20%** of your users are or could be french-speaking ones. **It is weird to notice that NOBODY in manjaro.fr community did not report it before.**

Sorry, I don't have time to fix a bug actived by your - sorry to be rude - ugly as hell theme. I have a lot of things to do, and be the _slave_ of xfce developers.

Thanks anyway for your complete answer here. I think this bug will be dead in your next default theme.

Have a good day.

**Comment by liberodark** [comment]
Have the same issue please fix this

**Comment by philmmanjaro** [comment]
The problem is with the **LANGUAGE**, used **PATH** to the picture, used **THEME** and **xfce-settings** itself when not used default folder `/usr/share/backgrounds/xfce` when not used having at least 30% of the letters `n-z` when there is a picture starting with the letters `a-m`. Simple put ...

As of now **FRENCH** has almost zero prio on my end, as Quality Management is done mostly in English and sometimes additionally in German from my end. I simply have no time to check other languages. And since only one language is affected it is what it is.

It is simply put a combination issue with that given language, which needs to be debugged from code side. Else I've proven to have a working workaround to avoid the issue when using **French**.

We might discuss it internally on how to proceed with this. As of now I won't put any further time into this silly issue for a language I don't use myself.

**Comment by liberodark** [comment]
Ok great just fix this whatever your solution
Thank you

**Comment by FredBezies** [comment]
My last answer of this issue:

> We might discuss it internally on how to proceed with this. As of now I won't put any further time into this silly issue for a language I don't use myself.


So, you're telling to your french users: we don't care about you, sorry. You'll have to wait for our next theme to get it fixed?

Strange way to deal with your users. Very weird. If you think it is the right one, well...

Have a good day anyway.

**Comment by philmmanjaro** [comment]
> So, you're telling to your french users: we don't care about you, sorry. You'll have to wait for our next theme to get it fixed?

I don't put effort into something, which only affects a language almost non of our team uses. It is a combination issue which now got reported. As it is a really low prio, we might fix it or not. Workarounds are given and therefore it is a known bug. Simply as that I've no time to deal with it. Other things are more important as fixing this right now.

**Comment by liberodark** [comment]
Just for new release don't add `breath-wallpaper` for new iso is a good deal and simple fix.

**Comment by FredBezies** [comment]
@philmmanjaro Thanks for your answer again. So if nobody speaks french in your team, it could be indeed hard to fix.

It is just a really bad advertisement for your distribution. That's all.

**Comment by philmmanjaro** [comment]
Double-Checked SX-Edition: Not present in v4.13 series of **XFCE**. Therefore no pressure in fixing it for v4.12 series.

**Comment by bignaux** [comment]
@FredBezies  is a known nasty troll. He just need a new target to write some shitty post on his website and social network...

**Comment by FredBezies** [comment]
@bignaux @philmmanjaro  Wenn man den Hund schlagen will, so hat er Leder gefressen. ; Give a dog a bad name, and hang him

Nothing more to say.

**Comment by oberon2007** [comment]
Practical summary.
This appears to be an upstream bug of soon to be deprecated xfce gtk2 version.
Since they have moved on to gtk3 with their development we cannot expect them to fix this in older versions.
On Manjaro level we don't really have a way to currently fix it. Next release will likely be xfce gtk3 anyway and it will no longer be relevant.
Available solutions for French systems for now:
* move wallpapers to **/usr/share/backgrounds/xfce** or create symlinks. Adjust default setting to use the new location.
* alternatively upgrade xfce packages to xfce-gtk3

**Comment by oberon2007** [comment]
In case we still have gtk2 packages in next release we can adjust settings-package to default to xfce directory.

**Comment by FredBezies** [comment]
A workaround to apply and test. Anyway it is a really weird bug :(

Thanks for the info.

**Comment by liberodark** [comment]
Thanks oberon :D

**Comment by philmmanjaro** [comment]
New **manjaro-xfce-settings** packages includes a [hook](https://github.com/manjaro/packages-community/blob/master/manjaro-xfce-settings/manjaro-xfce-settings.hook) to automatically sym-link added wallpapers from folder `/usr/share/backgrounds` to folder `/usr/share/backgrounds/xfce`. Dirs and already existing sym-links will be skipped. The user has to switch his default background folder to `/usr/share/backgrounds/xfce` on already installed systems.

This is a workaround and won't fix the given issues we have with folder `/usr/share/backgrounds` and combinations already described above.

**Comment by oberon2007** [comment]
Sounds good :+1: People just need ot be aware that as long as the default setting in **~/.config/xfce/fconf/xfce-perchannel-xml/xfce4-desktop.xml** will still crash the settings app. So the file should be removed before you try to adjust the wallpaper ...

**Comment by philmmanjaro** [comment]
Similar issues based on default folder choices:

https://bugs.launchpad.net/ubuntu/+source/xfdesktop4/+bug/1555951
https://bugs.launchpad.net/ubuntu/+source/xfdesktop4/+bug/1295614
https://bugzilla.xfce.org/show_bug.cgi?id=10759

**Comment by jonathonf** [comment]
This sounds like an issue that should be fixed upstream. Has it been reported there?

**Comment by oberon2007** [comment]
See my other [comment](https://github.com/manjaro/release-plan/issues/183#issuecomment-359970221)
It's most unlikely that xfce would fix a bug for older (gtk2) versions ...

**Comment by jonathonf** [comment]
OK, so this is an edge case when there is a very specific combination of language, theme, and path to a background image...

@FredBezies I don't think this is a "bad advertisement" for Manjaro at all. If noone on manjaro.fr has found the same issue then it's a very narrow issue to a maybe a couple of people. In general, if you want things to work you have to test them, otherwise noone will spot that there is an issue.

**Comment by FredBezies** [comment]
@jonathonf manjaro.fr knew it for months and DO NOTHING. See this thread: http://www.manjaro.fr/forum/viewtopic.php?f=23&t=8320 -> 10 mois = 10 months. Just use google translate.

They are useless for bug reporting. I had to shout on video to see something to happen.

**Comment by s7relok** [comment]
So a wontfix for this? As a french user, Manjaro can go f**k itself.
There is the bug, an few workarounds and solutions but a wontfix.

This thread is like a political debate, it's not my fault but my neighbour....

Just do your job and fix it. Or if you don't want to maintain distribution go do something else. It's not like french is a confidential language spoken by only a hundred persons in the world....

The managing of this bug is a joke, and the french official team is not better. Some french manjaro communities may have some hardass stubborn geeks, but they do the job. At least their ISO works for frenchs

**Comment by jonathonf** [comment]
> manjaro.fr knew it for months and DOES NOTHING

That's not an official forum, and if the bug isn't reported somewhere the maintainers read then of course it will do nothing. If the bug had been reported here it would at least have been seen. This is also a bug in Xfce rather than Manjaro, so a bug report to the Xfce project would have been the sensible thing to do rather than getting annoyed with people on a forum.

As it is, there's a fix released, e.g. see https://forum.manjaro.org/t/testing-update-2018-01-26-kernels-firefox-xfce-settings-gcc-mesa-haskell/39061

**Comment by FredBezies** [comment]
@jonathonf Of course it is not an official forum. But don't you think reporting on official forums or here on github **would have been** the **right thing to do** for months?

Just asking... I think - as a 11 years long mono-boot linux user - it is the thing to do. But hiding it from developers is a **shame**. Nothing less, nothing more.

I knew about the fix. But manjaro.fr even if it is not an official forum **is guilty** for not reporting this issue before.

Now that it is closed, let's get to something more interesting :)

**Comment by jonathonf** [comment]
> But don't you think reporting on official forums or here on github would have been the right thing to do for months?

Perhaps the people writing on the forum didn't think of it as serious enough to need reporting. You can't just go around blaming other people for not doing what you want them to. A video on a different site also isn't a great way to report a bug, so perhaps you did the same thing?

**Comment by FredBezies** [comment]
People doesn't think it was not serious enough? Really? It is just a basic feature you'll find in every single desktop environment. You can see this bug like "your nose in your face".

Blaming people? Really? You know what? I reported tons of bugs for a lot of projects., but as I know manjaro.fr did nothing **for months**, knowing that bug **was breaking user experience**, I had to shout. I don't like to be seen as the "bad boy".

I remembered some dark times back in 2017 related to manjaro project. I was not sure a bug report will be taken into account.

**Comment by s7relok** [comment]
That management at Manjaro seriously...

The #1 bug here s to not care with the appropriated attention to bugs
impacting directly end user.

To be a beginner friendly distro is not selling overpriced devices with the
system in it. It's to wipe what can annoy the average user, including bugs
that seems trivial

**Comment by jonathonf** [comment]
Dude... to fix a bug we have to know about a bug. Soon after it gained the attention of Phil it was fixed.

It's also a bug in Xfce not in Manjaro itself hence the request to file it with the upstream project and the initial "wontfix" tag.

**Comment by Ste74** [comment]
We have to hire an oracle to ask him lol.. We need a predictions

**Comment by oberon2007** [comment]
@s7relok
1. Bugs need to be reported in a way that developers know about them. GitHub is the best place for that.
2. We are not in a position to fix upstream xfce bugs
3. Since we know about the problem we have provide a workaround.
4. Manjaro doesn't sell hardware
3. The issue is not related to the Spitfire.
4. It's non of your business with whom developers decide to cooperate. You are crossing a line there. Disrespectful comments like this, be it here, in a video or wherever are neither appropriate nor productive.
5. This community is an environment of voluntary collaboration and mutual respect is required from contributing participants. You are not in a position to DEMAND anything from anyone here.

Contribute in a friendly and productive manner or go away! Comments like this are not welcome.

**Comment by Phoenix1747** [comment]
What the crap, some people here actually think that the moderators/owners of a random french (unofficial) manjaro forum automatically report all the bugs and problems that users experience?

That sort of bug is serious though you shouldn't expect the developers to crawl any random manjaro related post on the internet.

**Comment by FredBezies** [comment]
@oberon2007:

Just a little thing. Go there and look at the bottom of the page at http://manjaro.fr/forum/index.php

> © 2017 manjaro.fr ~ Communauté Francophone Manjaro 

Rough translation:

>  © 2017 manjaro.fr ~ French-speaking Manjaro Community

So, is it an official or a not official community? Both at the same time?

Anyway, this "community" knew about this bug for months... Normal or not to let it stay unreported?

@Phoenix: do you prefer a "What bug? Which bug?  There is no bug!" policy?

**Comment by Phoenix1747** [comment]
@FredBezies Dude these are normal people like you and me that work voluntarily in their freetime to build up a whole linux distro. What do you expect them to do? If you have a bug, report it on an official site, not any random community forum.

**Comment by Ste74** [comment]
In the main page of manjaro.org tag community you can see what are the officially forum.. Anyway is present also a mail list to talk with us.. So again only when the bug is reported in the proper way we can consider it..

**Comment by s7relok** [comment]
@oberon2007 Ok, so don't pretend you're a beginner friendly distro and be cocky about this.

1. Normal user don't know s**t about github and will never be. Yeah that's role of some to see what's happening in the communities. The name's community manager, and it not consist only on tweeting about releases and such. In addition, if you're so beginner friendly, at least multiply some easy means of communication for normal people and not geeky things to report problems.
2. You can workaround it easily apparently in the distro, stop passing the baby to another one when you're in trouble.
3. At least, because of geek blindness(github/irc/officialforum only) it took 10 months to resolve this...
4. You perfectly know what I want to say by "selling overpriced devices". That busy resources useful for.. dunno, resolving bugs or other tasks maybe? Or with the money hire people who can do the tasks if you are overflooded.
6. I give my user experience here. Not a contrib or dev one, because you can have the best team of the world, without users the project is nothing. Continue like this and the users will fly away. Mandriva if you don't know is the perfect example of a project with good potential killed by bad management and user pissing that fly somewhere else.
7. I'm as an (i think former counting of this evening) with some mjr installed rigs in the rights to ask a minimum of seriousness. I can understand the delay to find the bug, but considering it as a wontfix in first intention... are you serious? Don't you have the impression that the user who can't change it's wallpaper BY DEFAULT will consider the thing as a non-finished one?

To conclude : Continue to piss and ignoring user complains and you will see that distrowatch classification, like user interest, is not eternal

**Comment by Phoenix1747** [comment]
@FredBezies 
Well, if you don't like the people on the french forum then for god's sake use the official English forum. Dammit, manjaro.fr is an UNOFFICIAL COMMUNITY forum. Nobody there is responsible to report anything to the developers at all at any time.

**Comment by oberon2007** [comment]
@s7relok so you think you can tell me how I should spend my free time and energy or how to earn my money. That's totally insane.
Expect no further comment.

**Comment by FredBezies** [comment]
There is no love or hate here. 

People are fooled by manjaro.fr community name.

This bug was known for 10 months. So moderators of this unofficial forum did not take 5 minutes to report it? 

Was it too long to report it? I don't think so. 

French speaking users are fooled. And everybody pay that :(

Without my video on YouTube, I don't think both Oberon and Philm would have been aware of this bad bug.

To sum up? There is a lot of clean up to do :(

**Comment by Phoenix1747** [comment]
Nobody here can do anything about manjaro.fr, they are completely autonomous and can do whatever they want to do with their **community forum**. You should complain over there in their forum not here.

**Comment by s7relok** [comment]
@oberon2007 when you do a project it's a good thing to do it right. That remark and my comment earlier is not only for you, but for the team in general. 

And yes, your project acquired fame, and a userbase. Free to you to f**k everything up, but don't complain about users shouting and being not friendly. That's the reverse of the medal to be in a famous project. If you expect tranquility, I suggest Artix to you, seems to be more calm because the userbase is waaay littlier than MJR.

You don't seem to understand that is the reputation of the project and by extension the reputation of the persons working on it that is damaged by such things. And you're all like "that's not our problem, because it was not posted in the right platform". Unfortunately the bug is in YOUR iso (at least). And it needed many shouts and messages by french guys in order to lighten it to the official team, what a shame for a project who enlighten "user-friendly"

You all want to be "the ubuntu of arch" (simplified)? So give the project the means to do that. Or you'll be laughed by users.

**Comment by FredBezies** [comment]
Do you know the french expression : pisser dans un violon ?

It means: I don't care a... about what you say.

I know there is no need tout complain here. I was just thinking a little loud.

I think in the long run telling that a police like this site is the "enter-your-language" official  community for manjaro development team will be needed. 

Just a $0.02 idea ;)

**Comment by Ste74** [comment]
Well .. i'm tired .. we have IRC channel , twitter channel , forum channel , bug report platform , github and also mailing list to submit issue or talk about manjaro ! I think no need to continue this here. We are always with open mind .

**Comment by oberon2007** [comment]
@s7relok how dare you! Go back and read my comments in this thread. I have been factual, productive and trying to help solve your issue. Now you come here and insult me, tell me what to do and that I should leave the project? Who do you think you are?
FUCK OFF!

**Comment by philmmanjaro** [comment]
Ok, once again about the situation. Our communication language is only English. Period. We only do quality management on English and German. For all the other languages we simply can't pay attention at all.

We have members in our core team who speak Italian or Spanish. Also we translate the User-Guide to English and French, because our team member who is doing our documentation is from Canada. Since he uses mostly KDE, this issue was never to his attention. For screenshots he uses always the default theme we had decided to be used.

The French forum asked us if they can use the manjaro.fr domain and relate on our theming a little. That is all. Our team is ONLY on the forum.manjaro.org. Then we have a official German forum, which I hardly don't attend at all and that's it.

So only DUAL languages valid to be officially tested are ENGLISH and GERMAN. That's it. We had some bad blood about this issue and some general cliques some might have with this, but again, we simply have no time to do quality checks on languages non of our developers is using. However, we might fix issues, if we can on other languages, when best documented and we can follow instructions to reproduce.

We might find time to search for Manjaro in the internet based on english search terms, however won't crawl in any forums based on for us foreign languages to our developers. I'm and the team is really sorry for that.
